src/reducers/configuration.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+var validateConfiguration = require("../validate/validate-configuration");
+var isFunction = require("../utils/is-function");
+
+var initialState = {
+  httpAddresses: [],
+  wsAddresses: [],
+  ipcAddresses: [],
+  connectionTimeout: 3000,
+  pollingIntervalMilliseconds: 30000,
+  blockRetention: 100,
+  errorHandler: null
+};
+
+module.exports = function (configuration, action) {
+  var updatedConfiguration;
+  if (typeof configuration === "undefined") {
+    return initialState;
+  }
+  switch (action.type) {
+    case "SET_CONFIGURATION":
```

Comment:
If a user calls `SET_CONFIGURATION` not all of their configuration changes will occur.  The reason configuration was passed into `connect` previously was to ensure that configuration changes were applied everyone and to not give the user an impression that they could set configuration later and expect it to have an effect.

Recommend not storing configuration directly in Redux or updating this reducer to have side effects of resetting everything (reconnect, etc.).  I'm not personally a fan of the latter.